sstable/src/dictionary.rs

```diff
@@ -178,21 +225,35 @@ impl<TSSTable: SSTable> Dictionary<TSSTable> {
     /// Regardless of whether the term is found or not,
     /// the buffer may be modified.
     pub fn ord_to_term(&self, ord: TermOrdinal, bytes: &mut Vec<u8>) -> io::Result<bool> {
-        let mut sstable_reader = self.sstable_reader()?;
-        bytes.clear();
-        for _ in 0..(ord + 1) {
+        // find block in which the term would be
+        let Some(block_addr) = self.sstable_index.search_term_ordinal(ord) else {
+            return Ok(false)
+        };
+        let first_ordinal = block_addr.first_ordinal;
+
+        // then search inside that block only
+        let mut sstable_reader = self.sstable_reader_block(block_addr)?;
+        for _ in first_ordinal..(ord + 1) {
```

Comment:
nitpick... (=feel free to ignore)
```suggestion
        for _ in first_ordinal..=ord {
```